### PR TITLE
API provides functionality to finalize order

### PR DESCRIPTION
**Pull request by johnnysee**
PT: https://www.pivotaltracker.com/story/show/179716936

## User Story

```
As a restaurant API
In order for the user to complete their order
I would like to be able to update their existing order to be finalized and set the delivery time to finalized time + 30 minutes
```

## Tasks

- Create a PUT request to update the `finalized?` attribute
- Update carts table to include order status / Update Cart model with attribute `finalized?`:  true / false
- Update the response to include delivery time (time of update to `finalized` = true + 30 minutes
- Include the `finalized` property in the response
- Refactor the `carts_contoller` for readability
- Move the `custom_error`  method to `application_controller`

## Sub-tasks
Removed unused gem 'pry-byebug'

## Screenshot
Screenshot 2021-10-08 at 15 14 28

### CODE REVIEW

File: spec/requests/api/api_updates_cart_to_finalized_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            finalized: true,
+          },
+          headers: auth_headers
+    end
+
+    it { is_expected.to have_http_status 200 }
+
+    it 'is expected to update cart#finalized to "true"' do
+      cart.reload
+      expect(cart.finalized?).to eq true
```
Comment (tochman) [inline spec/requests/api/api_updates_cart_to_finalized_spec.rb]:
```
expect(cart.reload.finalized?).to eq true
```

File: Gemfile [hunk 2]
```diff
@@ -16,7 +16,6 @@ gem 'rack-cors', require: 'rack/cors'
 group :development, :test do
   gem 'simplecov', require: false
   gem 'pry-rails'
-  gem 'pry-byebug'
```
Comment (dorianbuck) [inline Gemfile:19]: There is no mention of removal of this gem in the PR description.

**Review by dorianbuck — COMMENTED** [review]
Great job guys!

File: Gemfile (hunk 2 above)
Comment (johnnysee) [reply Gemfile:19]: Done, added comment in PR, thank you Dorian!

**Pull request merged**